run-go-mod-tidy.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+if go mod tidy -v 2>&1 | grep -q 'updates to go.mod needed'; then
+    exit 1
+fi
+
+exec git add go.mod go.sum
```

Comment:
I believe it is bad practice to have a hook automatically `git add` changes. If I remember correctly the way pre-commit is supposed to work is that it tells you changes were made, and asks you to `git add` them yourself before committing again.

The reason for this is that it is best to have a human check the work before committing.

Reply:
I can force the hook to ask a human check instead, I just used the same way that [Husky](https://github.com/typicode/husky) did

Reply:
@dnephin I changed the script to notify user about `go.*` changements